From 124c5b72bb5c27c8caecba750eb44d62dc97fb68 Mon Sep 17 00:00:00 2001
From: jshe32X <jianchunx.shen@intel.com>
Date: Thu, 24 Oct 2013 12:57:09 +0800
Subject: [PATCH 06/17] SystemUI: Recycle the old wallpaper background bitmap
 after we choose a new wallpaper.

BZ: 101064

RootCause: ImageWallpaper didn't call old background bitmap's recycle before
           set it to null.

Category: aosp improvement
Domain: AOSP-Framework-Media
Origin: internal
Upstream-Candidate: yes

Orig-Change-Id: I27f6971a3edd26472b69e59b542b27fd7c8e7b90
Change-Id: Ice59aea79f8137d5995d7a5ce9a6ed7903750d30
Signed-off-by: jshe32X<jianchunx.shen@intel.com>
---
 .../SystemUI/src/com/android/systemui/ImageWallpaper.java    | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java b/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
index 50d862dee29..b1b622196c7 100644
--- a/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
+++ b/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
@@ -137,7 +137,10 @@ public class ImageWallpaper extends WallpaperService {
         @Override
         public void onDestroy() {
             super.onDestroy();
-            mBackground = null;
+            if (mBackground != null) {
+                mBackground.recycle();
+                mBackground = null;
+            }
             unloadWallpaper(true /* forgetSize */);
         }
 
-- 
2.17.1

